src/AutoRest.CSharp/Common/Generation/Writers/CodeWriterExtensions.Methods.cs

```diff
@@ -591,6 +592,27 @@ public static void WriteValueExpression(this CodeWriter writer, ValueExpression
                 case StringLiteralExpression(var literal, false):
                     writer.Literal(literal);
                     break;
+                case FormattableStringExpression(var formattable):
+                    writer.AppendRaw("$\"");
+                    var argumentCount = 0;
+                    foreach ((var span, bool isLiteral) in StringExtensions.GetPathParts(formattable.Format))
+                    {
+                        if (isLiteral)
+                        {
+                            writer.AppendRaw(span.ToString());
+                            continue;
+                        }
+
+                        var arg = formattable.GetArgument(argumentCount);
```

Comment:
This assumes that numbers in format are sequential, e.g.: `$"/subscriptions/{0}/resourceGroups/{1}/providers/Microsoft.Compute/exactMatchModel1s/{2}"`
Not sure if that is critical or not.

Reply:
Would it be sequential part of the rule when the formattable string is built?
For instance, `$"{foo}{bar}"` should be built into something like `string.Format("{0}{1}", foo, bar)`
Since this is happening internally, there is really no point that it could happen not in a sequence

Reply:
Or do you think that it is only reasonable the `FormattableStringExpression` should define like this:
```csharp
public record FormattableStringExpression(string Format, param ValueExpression[] args) : ValueExpression
```
the format here is just a string containing `{0}` like `some{0}else{1}`
and we might need to do some validation such as validating the number of `{x}` should be the same as the args
This could avoid the case that we might be passing in a arbitrary FormattableString into this expression with non-expression arguments (such as real string values) and we could only get the error when we write it.

Reply:
I changed the signature of FormattableStringExpression to the "format and args" way. There is a validation running on in debug mode.